ledger-tool/src/main.rs

```diff
@@ -2708,56 +2748,6 @@ fn main() {
                     },
                 );
 
-                accounts_index_config.index_limit_mb = if let Some(limit) =
-                    value_t!(arg_matches, "accounts_index_memory_limit_mb", usize).ok()
-                {
-                    IndexLimitMb::Limit(limit)
-                } else if arg_matches.is_present("disable_accounts_disk_index") {
-                    IndexLimitMb::InMemOnly
-                } else {
-                    IndexLimitMb::Unspecified
-                };
-
-                {
-                    let mut accounts_index_paths: Vec<PathBuf> =
-                        if arg_matches.is_present("accounts_index_path") {
-                            values_t_or_exit!(arg_matches, "accounts_index_path", String)
-                                .into_iter()
-                                .map(PathBuf::from)
-                                .collect()
-                        } else {
-                            vec![]
-                        };
-                    if accounts_index_paths.is_empty() {
-                        accounts_index_paths = vec![ledger_path.join("accounts_index")];
-                    }
-                    accounts_index_config.drives = Some(accounts_index_paths);
-                }
-
-                let filler_accounts_config = FillerAccountsConfig {
-                    count: value_t_or_exit!(arg_matches, "accounts_filler_count", usize),
-                    size: value_t_or_exit!(arg_matches, "accounts_filler_size", usize),
-                };
-
-                let accounts_db_config = Some(AccountsDbConfig {
-                    index: Some(accounts_index_config),
-                    accounts_hash_cache_path: Some(
-                        ledger_path.join(AccountsDb::ACCOUNTS_HASH_CACHE_DIR),
-                    ),
-                    filler_accounts_config,
-                    ancient_append_vec_offset: value_t!(
-                        matches,
-                        "accounts_db_ancient_append_vecs",
-                        i64
-                    )
```

Comment:
Another byproduct of this change is I discovered a bug with this arg (both verify and create-snapshot).

The logic is searching matches for `accounts_db_ancient_append_vecs`; however, `accounts_db_ancient_append_vecs` is an argument on the subcommand here:
https://github.com/solana-labs/solana/blob/ac7d0ff6331ed1376d39016bc500e99f1cfa22b1/ledger-tool/src/main.rs#L1793
and here
https://github.com/solana-labs/solana/blob/ac7d0ff6331ed1376d39016bc500e99f1cfa22b1/ledger-tool/src/main.rs#L1850

The match would actually be in `arg_matches` if it were present; thus, we'd never match even if the user passed it.

I have definitely burned myself on `matches` vs. `arg_matches` before; I think it is outside of the scope of this PR, but I might take a look at seeing if we can combine them or something to avoid this easy pitfall

Reply:
Decided to break out into https://github.com/solana-labs/solana/pull/29976 incase we want a backport.